src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -1616,63 +1626,93 @@ private function integerRangeMath(Type $range, BinaryOp $node, Type $operand): T
 				$min = $rangeMin !== null && $operand->getValue() !== 0 ? $rangeMin / $operand->getValue() : null;
 				$max = $rangeMax !== null && $operand->getValue() !== 0 ? $rangeMax / $operand->getValue() : null;
 			} else {
-				$min = $rangeMin !== null && $operand->getMin() !== null && $operand->getMin() !== 0 ? $rangeMin / $operand->getMin() : null;
-				$max = $rangeMax !== null && $operand->getMax() !== null && $operand->getMax() !== 0 ? $rangeMax / $operand->getMax() : null;
-			}
+				// Avoid division by zero when looking for the min and the max by using the closest int
+				$operandMin = $operandMin !== 0 ? $operandMin : 1;
+				$operandMax = $operandMax !== 0 ? $operandMax : -1;
+
+				if (
+					($operandMin < 0 || $operandMin === null)
+					&& ($operandMax > 0 || $operandMax === null)
+				) {
+					$result = TypeCombinator::union(
+						$this->integerRangeMath($range, $node, IntegerRangeType::fromInterval($operandMin, 0)),
+						$this->integerRangeMath($range, $node, IntegerRangeType::fromInterval(0, $operandMax))
+					)->toNumber();
+
+					if ($result->equals(new UnionType([new IntegerType(), new FloatType()]))) {
+						return new BenevolentUnionType([new IntegerType(), new FloatType()]);
+					} else {
+						return $result;
+					}
+				}
+				if (
+					($rangeMin < 0 || $rangeMin === null)
+					&& ($rangeMax > 0 || $rangeMax === null)
+				) {
+					$result = TypeCombinator::union(
+						$this->integerRangeMath(IntegerRangeType::fromInterval($rangeMin, 0), $node, $operand),
+						$this->integerRangeMath(IntegerRangeType::fromInterval(0, $rangeMax), $node, $operand)
+					)->toNumber();
+
+					if ($result->equals(new UnionType([new IntegerType(), new FloatType()]))) {
+						return new BenevolentUnionType([new IntegerType(), new FloatType()]);
+					} else {
+						return $result;
+					}
+				}
+
+				$rangeMinSign = ($rangeMin ?? -INF) <=> 0;
+				$rangeMaxSign = ($rangeMax ?? INF) <=> 0;
+
+				$min1 = $operandMin !== null ? ($rangeMin ?? -INF) / $operandMin : $rangeMinSign * -0.1;
+				$min2 = $operandMax !== null ? ($rangeMin ?? -INF) / $operandMax : $rangeMinSign * 0.1;
+				$max1 = $operandMin !== null ? ($rangeMax ?? INF) / $operandMin : $rangeMaxSign * -0.1;
+				$max2 = $operandMax !== null ? ($rangeMax ?? INF) / $operandMax : $rangeMaxSign * 0.1;
+
+				$min = min($min1, $min2, $max1, $max2);
+				$max = max($min1, $min2, $max1, $max2);
 
-			if ($range instanceof IntegerRangeType && $operand instanceof IntegerRangeType) {
-				if ($rangeMax === null && $operand->getMax() === null) {
-					$min = 0;
-				} elseif ($rangeMin === null && $operand->getMin() === null) {
+				if ($min === -INF) {
 					$min = null;
+				}
+				if ($max === INF) {
 					$max = null;
 				}
 			}
 
+			if ($min !== null && $max !== null && $min > $max) {
+				[$min, $max] = [$max, $min];
+			}
+
 			if ($operand instanceof IntegerRangeType
-				&& ($operand->getMin() === null || $operand->getMax() === null)
```

Comment:
The `float` type must be added as soon as I divide by a Range. For instance
```
int<-1, 2> / int<-1, 2>
```
There is `1 / 2` which is a float, even if both min and max are int.

Reply:
Strictly speaking, this is not always true.
like 
```
6 / int<1,3>
```
but I think it would get too complicated adding this logic in this PR, so no need to fix it now. It's not working correctly now anyway:)

Reply:
Yeah, I agree. I didn't find a better solution.
Should still be better than https://phpstan.org/r/e2efef83-28d5-4c57-83c6-c8be5fb4d67d

Moreover 6 / int<1, 6> is possibly a float.
So the only improvement could be to check every possible values unless we find a float...
This seems to complicated.

In the same way 6 / int<1,3> is `int<2, 6>` but the values 4 and 5 are not possible.
But again, unless testing all the values...

Reply:
Yeah, your solution is smart and the results are much better than before!
The number of edge cases are small because it only happens if 
range multiple (like 1 * 2 * 3) === constant integer (6) * k